### PR TITLE
Feature/186 vuetify profile page

### LINKED ISSUE #186: Update profile page to vuetify
**Description** 
- Looks good
- No more bootstrap

Relates to #144

### PR DESCRIPTION
**Pull request by TE4-oskar-pilborg**
## What issue did you implement or fix?
Fixed #186

## Summary
- Fixed profile.vue
- Worked with UsersModule.ts to make tests able to run and to make frontend easier to work with
- Removed sass code form navbar to make sites work properly  

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## How can this implementation be improved?
- These files could use update in documentation
- This will generate mergeconflicts with another pull request

### CODE REVIEW

File: frontend/src/store/modules/UsersModule.ts [hunk 1]
```diff
@@ -33,23 +33,28 @@ export interface UserState {
...
+    // }
   }
 
   @Action({rawError: true})
-  public fetchAll() {
+  public fetchAll(): Promise<object> {
     return new Promise((resolve, reject) => {
       UsersAPI.all()
         .then((response: User[]) => {
+          console.log(response);
```
Comment (te4-hugo-svanberg) [inline frontend/src/store/modules/UsersModule.ts]: Is this console.log necessary?

Comment (TE4-oskar-pilborg) [reply frontend/src/store/modules/UsersModule.ts]: nope

File: frontend/src/components/GoogleLogin.vue [hunk 2]
```diff
@@ -6,7 +6,7 @@
       href="#"
       @click="signOut"
     >SIGN OUT</a>
-    <div id="signin2" />
+    <div v-else id="signin2" />
```
Comment (te4-gabriel-stark) [inline frontend/src/components/GoogleLogin.vue]: You dont need this in the new and improved NavbarComponent B)
Also the stuff above that one is not needed in the new and improved NavbarComponent B)
```suggestion
    <div id="signin2" />
```

File: frontend/src/store/modules/LoansModule.ts [hunk 3]
```diff
@@ -99,7 +99,7 @@ class LoansModule extends VuexModule {
 
   @Mutation
   private setFailure(payload: any) {
-    this.loanState.failure = payload;
+    // this.loanState.failure = payload;
```
Comment (te4-gabriel-stark) [inline frontend/src/store/modules/LoansModule.ts]: Now that this function is empty couldn't you remove it from the face of the earth?

File: frontend/src/store/modules/UsersModule.ts [hunk 4]
```diff
@@ -33,20 +33,24 @@ export interface UserState {
 class UsersModule extends VuexModule {
   private userState: UserState = {users: {}, current_user: null, failure: null};
 
-  get all() {
+  get currentUserID(): number {
+    return this.userState.current_user;
```
Comment (te4-gabriel-stark) [inline frontend/src/store/modules/UsersModule.ts]: Aren't we supposed to use camelCase in the ts files?

File: frontend/src/store/modules/UsersModule.ts [hunk 5]
```diff
@@ -33,20 +33,24 @@ export interface UserState {
...
-      return this.all[this.userState.current_user];
-    } else {
-      return {};
-    }
+  get currentUser(): User {
+    // if (this.userState && this.userState.current_user) {
+    return this.all[this.userState.current_user];
+    // } else {
+    //   return null;
+    // }
```
Comment (te4-gabriel-stark) [inline frontend/src/store/modules/UsersModule.ts]: Can you remove these comments?

File: frontend/src/store/modules/UsersModule.ts [hunk 6]
```diff
@@ -107,24 +111,25 @@ class UsersModule extends VuexModule {
...
+    Vue.set(this.userState, 'current_user', payload.uid);
   }
 
   @Mutation
-  public setFailure(payload: any) {
-    this.userState.failure = payload;
+  public setFailure(payload: any): void {
+    // console.log(payload);
+    // this.userState.failure = payload;
   }
```
Comment (te4-gabriel-stark) [inline frontend/src/store/modules/UsersModule.ts]: Couldn't you remove this function, seeing that it doesn't do anything?

File: frontend/src/components/GoogleLogin.vue (hunk 2 above)
Comment (TE4-oskar-pilborg) [reply frontend/src/components/GoogleLogin.vue]: okay nice

File: frontend/src/store/modules/LoansModule.ts (hunk 3 above)
Comment (TE4-oskar-pilborg) [reply frontend/src/store/modules/LoansModule.ts]: sure could

File: frontend/src/store/modules/UsersModule.ts (hunk 4 above)
Comment (TE4-oskar-pilborg) [reply frontend/src/store/modules/UsersModule.ts]: yes, it did just not occur to me due to the name being old

**Pull request merged**